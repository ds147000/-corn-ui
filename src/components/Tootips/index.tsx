import { View } from '@tarojs/components'
import React from 'react'



export declare namespace TOOTIPS {
  type Align = 'top' | 'bottom' | 'left' | 'right'

  type Item = { text: string; type: 'active' | 'defualt' }

  interface Props {
    align: Align
    list: Item[]
    onClick?(item: Item, index: number): void
  }

}


const TootipsCmps: React.FC<TOOTIPS.Props> = () => {
  return (
    <View />
  )
}

export default TootipsCmps
