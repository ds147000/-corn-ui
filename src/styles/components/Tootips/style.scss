.xrk-tooltip {
  background: $white;
  box-shadow: 0 0 50px 0 #f3f5fa;
  text-align: center;
  border-radius: 15px;
  width: 180px;
  position: absolute;
  z-index: $portalIndex + 1;

  &-top {
    transform: translate(0 , -100%);
  }

  &-arrow {
    display: block;
    border-style: solid;
    position: absolute;
    z-index: 1;

    &-bottom {
      border-top-width: 6px;
      border-left-width: 10px;
      border-right-width: 10px;
      border-bottom-width: 6px;
      border-top-color: rgba($color: $white , $alpha: 0);
      border-right-color: rgba($color: $white , $alpha: 0);
      border-left-color: rgba($color: $white , $alpha: 0);
      border-bottom-color: rgba($color: $white , $alpha: 1);
    }

    &-top {
      border-top-width: 6px;
      border-left-width: 10px;
      border-right-width: 10px;
      border-bottom-width: 6px;
      border-top-color: rgba($color: $white , $alpha: 1);
      border-right-color: rgba($color: $white , $alpha: 0);
      border-left-color: rgba($color: $white , $alpha: 0);
      border-bottom-color: rgba($color: $white , $alpha: 0);
    }
  }

  &-item {
    width: 180px;
    height: 92px;
    font-size: 30px;
    font-weight: 400;
    color: $text1;
    white-space: nowrap;
    text-overflow: ellipsis;
    overflow: hidden;
  }

  &-active {
    font-size: 30px;
    font-weight: bolder;
    color: $error;
  }
}
