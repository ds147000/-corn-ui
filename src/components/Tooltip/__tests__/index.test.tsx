import React from 'react'
import { fireEvent, render, waitFor } from '@testing-library/react'
import Tootips, { TOOTIPS } from '../index'

describe('Tootips', () => {
  test('快照', async () => {
    const list: TOOTIPS.Item[] = [
      { text: '金币', type: 'defualt' },
      { text: '人鱼', type: 'active' }
    ]
    const screen = render(<Tootips align="bottom" list={list}>更多</Tootips>)

    await waitFor(() => expect(screen).toMatchSnapshot())
  })

  test('open', async () => {
    const list: TOOTIPS.Item[] = [
      { text: '金币', type: 'defualt' },
      { text: '人鱼', type: 'active' }
    ]
    const screen = render(<Tootips align="bottom" list={list}>更多</Tootips>)
    fireEvent.click(screen.getByText('更多'))
    await waitFor(() => expect(screen.queryByText('金币')).not.toBeNull())
    await waitFor(() => expect(screen.queryByText('人鱼')).not.toBeNull())
    await waitFor(() => expect(screen).toMatchSnapshot())
  })

  test('open move', async () => {
    const list: TOOTIPS.Item[] = [
      { text: '金币', type: 'defualt' },
      { text: '人鱼', type: 'active' }
    ]
    const screen = render(<Tootips align="bottom" moveX={10} moveY={50} list={list}>更多</Tootips>)
    fireEvent.click(screen.getByText('更多'))
    await waitFor(() => expect(screen.queryByText('金币')).not.toBeNull())
    await waitFor(() => expect(screen.queryByText('人鱼')).not.toBeNull())
    await waitFor(() => expect(screen).toMatchSnapshot())
  })

  test('click', async () => {
    const list: TOOTIPS.Item[] = [
      { text: '金币', type: 'defualt' },
      { text: '人鱼', type: 'active' }
    ]
    const onClick = jest.fn()
    const screen = render(<Tootips list={list} onClick={onClick} >更多</Tootips>)
    fireEvent.click(screen.getByText('更多'))
    await waitFor(() => expect(screen.queryByText('金币')).not.toBeNull())
    await waitFor(() => expect(screen.queryByText('人鱼')).not.toBeNull())
    fireEvent.click(screen.getByText('金币'))
    await waitFor(() => expect(onClick).toHaveBeenCalledTimes(1))
    await waitFor(() => expect(onClick.mock.calls[0][1]).toBe(0))
  })
})
