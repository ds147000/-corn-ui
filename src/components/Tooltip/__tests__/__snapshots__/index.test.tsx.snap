// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tootips open 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="xrk-popover-plane"
        data-testid="mask"
      />
      <div
        class="xrk-popover-show"
        data-testid="centent"
      >
        <div
          class="xrk-tooltip-arrow xrk-tooltip-arrow-bottom"
          style="top: 0px; left: 0px;"
        />
        <div
          class="xrk-tooltip xrk-tooltip-bottom"
          style="top: 8.533333333333333px; left: -76.80000000000001px;"
        >
          <div
            class="xrk-tooltip-item xrk-f xrk-ac xrk-jc"
          >
            金币
          </div>
          <div
            class="xrk-tooltip-item xrk-f xrk-ac xrk-jc xrk-tooltip-active"
          >
            人鱼
          </div>
        </div>
      </div>
      <div
        class="xrk-popover "
      >
        更多
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="xrk-popover-plane"
      data-testid="mask"
    />
    <div
      class="xrk-popover-show"
      data-testid="centent"
    >
      <div
        class="xrk-tooltip-arrow xrk-tooltip-arrow-bottom"
        style="top: 0px; left: 0px;"
      />
      <div
        class="xrk-tooltip xrk-tooltip-bottom"
        style="top: 8.533333333333333px; left: -76.80000000000001px;"
      >
        <div
          class="xrk-tooltip-item xrk-f xrk-ac xrk-jc"
        >
          金币
        </div>
        <div
          class="xrk-tooltip-item xrk-f xrk-ac xrk-jc xrk-tooltip-active"
        >
          人鱼
        </div>
      </div>
    </div>
    <div
      class="xrk-popover "
    >
      更多
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Tootips open move 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="xrk-popover-plane"
        data-testid="mask"
      />
      <div
        class="xrk-popover-show"
        data-testid="centent"
      >
        <div
          class="xrk-tooltip-arrow xrk-tooltip-arrow-bottom"
          style="top: 0px; left: 0px;"
        />
        <div
          class="xrk-tooltip xrk-tooltip-bottom"
          style="top: 18.53333333333333px; left: -26.80000000000001px;"
        >
          <div
            class="xrk-tooltip-item xrk-f xrk-ac xrk-jc"
          >
            金币
          </div>
          <div
            class="xrk-tooltip-item xrk-f xrk-ac xrk-jc xrk-tooltip-active"
          >
            人鱼
          </div>
        </div>
      </div>
      <div
        class="xrk-popover "
      >
        更多
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="xrk-popover-plane"
      data-testid="mask"
    />
    <div
      class="xrk-popover-show"
      data-testid="centent"
    >
      <div
        class="xrk-tooltip-arrow xrk-tooltip-arrow-bottom"
        style="top: 0px; left: 0px;"
      />
      <div
        class="xrk-tooltip xrk-tooltip-bottom"
        style="top: 18.53333333333333px; left: -26.80000000000001px;"
      >
        <div
          class="xrk-tooltip-item xrk-f xrk-ac xrk-jc"
        >
          金币
        </div>
        <div
          class="xrk-tooltip-item xrk-f xrk-ac xrk-jc xrk-tooltip-active"
        >
          人鱼
        </div>
      </div>
    </div>
    <div
      class="xrk-popover "
    >
      更多
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Tootips 快照 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="xrk-popover "
      >
        更多
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="xrk-popover "
    >
      更多
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
